tap_sailthru/streams.py

```diff
@@ -467,38 +405,85 @@ def prepare_request_payload(
         Returns:
             A dictionary containing the request payload.
         """
-        return {'primary': 1}
+        return {"primary": 1}
 
     def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
-        return {
-            "list_id": record["list_id"],
-            "list_name": record["name"]
-        }
+        return {"list_id": record["list_id"], "list_name": record["name"]}
 
     def parse_response(
-        self,
-        response: SailthruResponse,
-        context: Optional[dict]
+        self, response: SailthruResponse, context: Optional[dict]
     ) -> Iterable[dict]:
         """Parse the response and return an iterator of result rows."""
-        for row in response['lists']:
-            row['account_name'] = self.config.get('account_name')
+        for row in response["lists"]:
+            row["account_name"] = self.config.get("account_name")
             yield row
 
 
+class ListMembersParentStream(sailthruStream):
+    """Manufactured Stream to help with load balancing for List Members stream"""
+
+    name = "list_members_parent"
+    schema_filepath = SCHEMAS_DIR / "lists.json"
+
+    def request_records(self, context: Optional[dict]) -> Iterable[dict]:
+        lists_query = (
+            """
+            SELECT
+                id as list_id,
+                list_name,
+                max(valid_email_count) email_count
+            FROM `"""
+            + self.config.get("table_id")
+            + """`
+            where
+                account = '"""
+            + self.config.get("account_name")
+            + """'
+                and is_primary
+            group by 1, 2
+            order by 2
+        """
+        )
+        self.logger.info(f"Executing query: {lists_query}")
+        client = bigquery.Client(project="nymag-analytics-157315")
```

Comment:
Should we move the project ID also to config? Or alternatively this might work without the project ID too since we have `GOOGLE_APPLICATION_CREDENTIALS` set in prod